datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -577,19 +578,14 @@ fn coerce_arguments_for_fun(
 
     let mut expressions: Vec<Expr> = expressions.to_vec();
 
-    // Cast Fixedsizelist to List for array functions
-    if *fun == BuiltinScalarFunction::MakeArray {
+    // coerce the fixed size list to list for all array fucntions
+    if fun.name().contains("array") {
```

Comment:
There are two problems that I think we should stick to the previous checking.
1. name() takes the first aliases of the function, so if there we forgot the have first aliases starts with array, it fails.
2. We just need to care about `make_array` only, then other functions should get non-fixedsizelsit as input.

Reply:
Why do we only need to care about `make_array`? If we put `coerce_arguments_for_fun` in front of `coerce_arguments_for_signature`, it should cast all `FixedSizeList` to `List` regarding array functions.

Reply:
Because any kind of array functions includes make_array(). Like `array_append(cast(fixedsizelist, list), element)`. we dont need to check again with `array_append`.

Reply:
Ok, I know the reason, it seems we still need to handle fixedsizelist for other array functions because we can pass Fixedsizelist directly to them with `arrow_cast`.

In this case, I agree we should handle them in `coerce_arguments_for_signature`. And, `coerce_arguments_for_fun` can be removed.

Reply:
We can keep `coerce_arguments_for_fun ` until all `FixedSizeList` type coercion work is done.